### PR TITLE
Fix minstake

### LINKED ISSUE #872: Test MinStakePerDelegate
Test that if any of 5 provider's stake pool is less than `MinStakePerDelegate` setting it doesn't receive rewards

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  | fix/minstake  |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

File: code/go/0chain.net/smartcontract/minersc/miner.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ func (msc *MinerSmartContract) AddMiner(t *transaction.Transaction,
 
 	lockAllMiners.Lock()
 	defer lockAllMiners.Unlock()
+
+	logging.Logger.Info("Jayash add_miner: The new miner info", zap.Any("minStakePerDelegate", gn.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/miner.go]: removre

File: code/go/0chain.net/smartcontract/minersc/models.go [hunk 2]
```diff
@@ -271,6 +271,7 @@ func (gn *GlobalNode) readConfig() (err error) {
 		return
 	}
 	gn.MinStakePerDelegate, err = currency.ParseZCN(config2.SmartContractConfig.GetFloat64(pfx + SettingName[MinStakePerDelegate]))
+	logging.Logger.Info("Jayash MinStakePerDelegate", zap.Any("MinStakePerDelegate", gn.MinStakePerDelegate), zap.Any("config", config2.SmartContractConfig.GetFloat64(pfx+SettingName[MinStakePerDelegate])))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/models.go]: remove

File: code/go/0chain.net/smartcontract/minersc/models.go [hunk 3]
```diff
@@ -380,6 +381,7 @@ func (gn *GlobalNode) Get(key Setting) (interface{}, error) {
 	case MinStake:
 		return gn.MinStake, nil
 	case MinStakePerDelegate:
+		logging.Logger.Info("Jayash get", zap.Any("MinStakePerDelegate", gn.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/models.go]: remove

File: code/go/0chain.net/smartcontract/minersc/settings.go [hunk 4]
```diff
@@ -232,6 +234,7 @@ func (gn *GlobalNode) setBalance(key string, change currency.Coin) error {
 	case MinStake:
 		gn.MinStake = change
 	case MinStakePerDelegate:
+		logging.Logger.Info("Jayash setBalance", zap.Any("change", change))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/settings.go]: remove

File: code/go/0chain.net/smartcontract/stakepool/stakepool.go [hunk 5]
```diff
@@ -390,6 +390,7 @@ func (sp *StakePool) DistributeRewardsRandN(
 		return err
 	}
 
+	logging.Logger.Info("Jayash DistributeRewardsRandN", zap.Any("Minstake", sp.Settings.MinStake), zap.Any("total", total), zap.Any("value", value), zap.Any("randN", randN))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/stakepool/stakepool.go]: remove

File: code/go/0chain.net/smartcontract/storagesc/config.go [hunk 6]
```diff
@@ -332,6 +334,7 @@ func getConfiguredConfig() (conf *Config, err error) {
 		return nil, err
 	}
 	conf.MinStakePerDelegate, err = currency.ParseZCN(scc.GetFloat64(pfx + "min_stake_per_delegate"))
+	logging.Logger.Info("Jayash getConfiguredConfig", zap.Any("MinStakePerDelegate", conf.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/config.go]: remove

File: code/go/0chain.net/smartcontract/storagesc/config_settigns.go [hunk 7]
```diff
@@ -393,6 +395,7 @@ func (conf *Config) setCoin(key string, change currency.Coin) error {
 	case MinStake:
 		conf.MinStake = change
 	case MinStakePerDelegate:
+		logging.Logger.Info("Jayash setting min stake per delegate", zap.Any("change", change))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/config_settigns.go]: remove

File: code/go/0chain.net/smartcontract/storagesc/config_settigns.go [hunk 8]
```diff
@@ -640,6 +643,7 @@ func (conf *Config) get(key Setting) interface{} {
 	case MinStake:
 		return conf.MinStake
 	case MinStakePerDelegate:
+		logging.Logger.Info("Jayash getting min stake per delegate", zap.Any("conf", conf.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/config_settigns.go]: remove

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 9]
```diff
@@ -324,6 +326,7 @@ func (ssc *StorageSmartContract) getOrCreateStakePool(
 
 	sp.Settings.ServiceChargeRatio = settings.ServiceChargeRatio
 	sp.Settings.MaxNumDelegates = settings.MaxNumDelegates
+	logging.Logger.Info("Jayash getOrCreateStakePool", zap.Any("MinStakePerDelegate", conf.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go]: remove

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 10]
```diff
@@ -341,6 +344,7 @@ func (ssc *StorageSmartContract) createStakePool(
 	sp.Minter = chainstate.MinterStorage
 	sp.Settings.ServiceChargeRatio = settings.ServiceChargeRatio
 	sp.Settings.MaxNumDelegates = settings.MaxNumDelegates
+	logging.Logger.Info("Jayash 2getOrCreateStakePool", zap.Any("MinStakePerDelegate", conf.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go]: reemove

File: code/go/0chain.net/smartcontract/zcnsc/stakepool.go [hunk 11]
```diff
@@ -137,6 +139,9 @@ func (zcn *ZCNSmartContract) getOrUpdateStakePool(gn *GlobalNode,
 		sp.Settings.MaxNumDelegates = settings.MaxNumDelegates
 		changed = true
 	}
+
+	logging.Logger.Info("Jayash getOrCreateStakePool", zap.Any("MinStakePerDelegate", gn.MinStakePerDelegate))
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/zcnsc/stakepool.go]: remove

**Review by dabasov — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**